packages/mobile/src/services/offline-downloader/offline-storage.ts

```diff
@@ -70,21 +71,42 @@ export const getTrackJson = async (
   }
 }
 
-export const verifyTrack = async (trackId: string): Promise<boolean> => {
-  const audioFile = exists(getLocalAudioPath(trackId))
+export const writeTrackJson = async (
+  trackId: string,
+  trackToWrite: UserTrackMetadata
+) => {
+  const pathToWrite = getLocalTrackJsonPath(trackId)
+  if (await exists(pathToWrite)) {
+    await RNFS.unlink(pathToWrite)
+  }
+  await RNFS.write(pathToWrite, JSON.stringify(trackToWrite))
+}
+
+export const verifyTrack = async (
+  trackId: string,
+  expectTrue?: boolean
+): Promise<boolean> => {
+  const audioFile = await exists(getLocalAudioPath(trackId))
   // TODO: check for all required art
-  // const artFile = exists(path.join(getLocalTrackDir(trackId), '150x150.jpg'))
+  // const artFile = await exists(
+  //   path.join(getLocalTrackDir(trackId), '150x150.jpg')
+  // )
   const artFile = true
-  const jsonFile = exists(getLocalTrackJsonPath(trackId))
+  const jsonFile = await exists(getLocalTrackJsonPath(trackId))
 
   const results = await Promise.allSettled([audioFile, artFile, jsonFile])
-  const [audioExists, artExists, jsonExists] = results
+  const booleanResults = results.map(
```

Comment:
maybe instead of a map -> every, we just return every and in that do the stuff we did in map?

Reply:
that would be cleaner, but we won't know inside the map which promise result is which, so we can't log the appropriate message. I could bind the name to each promise, but I think that will result in less readable code